jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/IRabitTracker.java

```diff
@@ -0,0 +1,20 @@
+package ml.dmlc.xgboost4j.java;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface for Rabit tracker implementations with three public methods:
+ *
+ *  - start(): Start the rabit tracker awaiting for worker connections.
+ *  - getWorkerEnvs(): Return the environment variables needed to initialize Rabit clients.
+ *  - waitFor(): Wait for the task execution by the worker nodes.
+ */
+public interface IRabitTracker {
+  Map<String, String> getWorkerEnvs();
+  boolean start();
+  // for Scala-implementation compatibility
+  boolean start(long timeout, TimeUnit unit);
+  int waitFor();
+  int waitFor(long timeout, TimeUnit unit);
```

Comment:
Really? My personal experience with loading RDDs and then training is that `Rabit.init(rabitEnv.asJava)` runs after loading the RDDs from HDFS, because how Spark constructs the stages as DAGs. I could be wrong though. My experience was based on the log files: the Rabit tracker didn't show any worker connections until all the RDD loading stages are finished. Maybe it was because I added many intermediate stages between data loading and training: I used customized LZO-compressed libsvm file loader to load data, and then use a stratified sampler to transform the RDD, before calling `train()` on XGBoost. 

Okay, I can remove that constructor argument, and make it a hard-coded value of 5 minutes or something.